### PR TITLE
Fixes, tests, and more whitelisting for the Noodle security sandbox

### PR DESCRIPTION
**Pull request by jmuscireum**

### CODE REVIEW

File: src/main/java/sirius/pasta/noodle/sandbox/Sandbox.java [hunk 1]
```diff
@@ -96,21 +98,64 @@ public static boolean isAccessGranted(NoodleSandbox annotation) {
...
+        if (superclass == null) {
+            return null;
+        }
+
+        try {
+            return superclass.getMethod(method.getName(), method.getParameterTypes());
+        } catch (NoSuchMethodException e) {
+            Exceptions.ignore(e);
+            return null;
+        }
```
Comment (andyHa) [inline src/main/java/sirius/pasta/noodle/sandbox/Sandbox.java:137]: what if an intermediate class doesn't provide the method? will this still work or does getMethod only provide the methods defined of the class itself

File: src/main/resources/component-web.conf [hunk 2]
```diff
@@ -411,18 +411,41 @@ scripting {
...
             "sirius.kernel.commons.Value.*" = true
             "sirius.kernel.commons.Amount.*" = true
             "sirius.kernel.commons.Strings.*" = true
+            "sirius.kernel.settings.ExtendedSettings.getExtension" = true
             "sirius.kernel.settings.Settings.get" = true
+            "sirius.kernel.settings.Settings.getSettings" = true
+            "sirius.kernel.settings.Settings.getString" = true
+            "sirius.kernel.settings.Settings.getStringList" = true
+            "sirius.kernel.settings.Settings.getTranslatedString" = true
+            "sirius.kernel.Sirius.getSettings" = true
```
Comment (andyHa) [inline src/main/resources/component-web.conf:439]: I'd not reveal the complete system config to a sandboxed user - there is way too much interesting stuff in there...

**Review by andyHa — CHANGES_REQUESTED** [review]

File: src/main/resources/component-web.conf (hunk 2 above)
Comment (jmuscireum) [reply src/main/resources/component-web.conf]: Yeah, of course. That's what reviews are for 😅

File: src/main/java/sirius/pasta/noodle/sandbox/Sandbox.java (hunk 1 above)
Comment (jmuscireum) [reply src/main/java/sirius/pasta/noodle/sandbox/Sandbox.java:137]: It works as intended, getMethod finds the first match in the inheritance tree

File: src/main/java/sirius/pasta/noodle/sandbox/Sandbox.java [hunk 3]
```diff
@@ -96,21 +98,64 @@ public static boolean isAccessGranted(NoodleSandbox annotation) {
...
+        Class<?> superclass = method.getDeclaringClass().getSuperclass();
+
+        if (superclass == null) {
+            return null;
+        }
+
+        try {
+            return superclass.getMethod(method.getName(), method.getParameterTypes());
+        } catch (NoSuchMethodException e) {
+            Exceptions.ignore(e);
```
Comment (sabieber) [inline src/main/java/sirius/pasta/noodle/sandbox/Sandbox.java:135]:
```suggestion
        } catch (NoSuchMethodException exception) {
            Exceptions.ignore(exception);
```

**Review by sabieber — APPROVED** [review]

**Review by idlira — APPROVED** [review]

**Review by mkeckmkeck — APPROVED** [review]

**Review by andyHa — APPROVED** [review]

**Pull request merged**